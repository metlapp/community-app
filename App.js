import React, { useEffect } from "react";
import { StyleSheet, View } from "react-native";
import LoginScreen from "./screens/LoginScreen";
import AuthContext from "./auth/Context";
import AccountScreen from "./screens/AccountScreen";
import { Provider as PaperProvider } from "react-native-paper";
import authStorage from "./auth/Storage";

export default function App() {
  const [user, setUser] = React.useState();

  const restoreUser = async () => {
    const userData = await authStorage.getUser();

    //If no user data is in the storage than just return which shows the login screen
    if (!userData) {
      setUser();
      return;
    }
    setUser(userData);
  };

  //Checks the storage on open of the app
  useEffect(() => {
    restoreUser();
  }, []);

  return (
    <AuthContext.Provider value={{ user, setUser }}>
      <PaperProvider>
        <View>{user ? <AccountScreen /> : <LoginScreen />}</View>
      </PaperProvider>
    </AuthContext.Provider>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});
