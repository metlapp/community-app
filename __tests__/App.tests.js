import "react-native";
import React from "react";
import { shallow } from "enzyme";
import App from "../App";

// Note: this is just for use with Jest snapshot testing
// and comes packaged with react-native init project.
// You do not need this if using Enzyme 'toMatchSnapshot' etc.
import renderer from "react-test-renderer";

// This test just uses Jest snapshot testing
it("renders correctly, test using Jest", () => {
  renderer.create(<App />);
});

// Using Jest + Enzyme
describe("<App />", () => {
  it("renders correctly, test using Jest + Enzyme", () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
  it("matches the snapshot", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("text on homepage", () => {
    const wrapper = shallow(<App />);
    const text = wrapper.find("Text").render().text();
    expect(text).toEqual("Open up App.js to start working on your app!");
  });
});
