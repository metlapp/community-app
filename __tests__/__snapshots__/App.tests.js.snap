// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot 1`] = `null`;

exports[`<App /> renders correctly, test using Jest + Enzyme 1`] = `
<ContextProvider
  value={
    Object {
      "setUser": [Function],
      "user": undefined,
    }
  }
>
  <Provider>
    <ForwardRef(NavigationContainer)>
      <AuthNavigator
        test={false}
      />
    </ForwardRef(NavigationContainer)>
  </Provider>
</ContextProvider>
`;
exports[`<App /> renders correctly, test using Jest + Enzyme 1`] = `""`;
